_topics/en/pooled-mining.md

```diff
@@ -0,0 +1,216 @@
+
+- The coinbase transaction must follow a template provided by the pool.
+  For example, all pools we know of use the coinbase transaction to pay
+  either the pool operator (who later distributes rewards to pool
+  members) or to pay pool members directly.
+
+Pools may impose additional requirements on their members, either
+explicitly or by using a pool protocol that does not give members the
+flexibility to make certain choices.
+
+## Block withholding attacks
+
+Pooled mining is fundamentally vulnerable to a problem knows as a _block
+withholding attack_.  There are two types of shares: those that have
+enough work to be a valid block and those that don't.  In a pure
+pay-per-share (PPS) scheme, the pool pays the same amount for both types
+but the pool only earns rewards from the Bitcoin system for producing
+valid blocks.  This can allow a pool member to forfeit a small amount of
+income in order to deny the pool a large amount of income.
+
+For example, image the ratio between the primary target and secondary
+target is 1,000, so for every 1,000 shares submitted by pool members,
+the pool would be expected to find one block on average.  Malicious
+pool member Mallory submits each share she finds except those that have
+enough proof of work to be valid blocks.  On average, Mallory gets
+paid for 999 blocks out of every 1,000 she generates, reducing her
+income by 0.1% but costing the pool 100% of a block reward for every
+1,000 shares generated by Mallory.
+
+We are unaware of any deployed method that fully addresses this problem.
+A common mitigation is for pools to pay more for shares that have enough
+work to be a valid block (which implies paying less for other shares).
+Some pools also restrict who can join their pools and use techniques
+from statistical analysis to attempt to detect block withholders.
+
+## Pool protocols
+
+The earliest pool protocol was built around transmitting block header
+templates in the format used by the early Bitcoin Core RPC `getwork`.
+When the speed of mining equipment made that impractical, an early
+attempt at creating a distributed mining protocol was specified in BIPs
+[22][BIP22] and [23][BIP23], with a partial implementation in Bitcoin
+Core as the `getblocktemplate` RPC.  GetBlockTemplate never saw
+widespread usage as a pool protocol itself, although many centralized
+pool servers have used it for the past decade in the background to obtain
+a good set of transactions to mine from their local Bitcoin Core mode.
+
+As the GetWork-based protocol became problematic, most mining pools
+switched to the Stratum protocol (v1).  Stratum v1 allows pools to
+only send their members a template for a block header and a coinbase
+transaction, which can be less than 200 bytes.  When a member finds a
+share, they can send back roughly the same information.  This very
+compact information minimizes the overhead of the pool protocol.  The
+downside of the way GetBlock and Stratum v1 are typically used is that
```

Comment:
GetBlockTemplate or GetWork?

```suggestion
downside of the way GetBlock and Stratum v1 are typically used is that
```

Reply:
GetBlock is correct.  If you received GetBlockTemplate from your pool server, you'd have a complete list of every transaction in the template, so blind mining wouldn't be a concern.  (But AFAIK no pool server has ever sent GBT to its members; that would be bandwidth intensive---GBT can be 2x the 3x the size of the candidate block itself.)